### DISCUSSION TITLE
Why deepPartial my required formValue?

### DISCUSSION THREAD

**Discussion by tianzhich** [react-hook-form/react-hook-form#3694] [Q&A]
I have a case in which I defined a required field of my form. And I passed type definition to `useForm`:

```tsx
interface FormValue {
  id: string;
}

function MyForm() {
  const { reset } = useForm<FormValue>({ defaultValues: { id: 'foo' } });
}
```

But the type of `reset` function's param was inferred as `{ id?: string }`. So I lost type prompt even when i forgot setting `id`.

Is this an intended behavior? And why if it is?

**Comment by tianzhich** [comment]
If my field is an `Object`. I just think it should never be `undefined` so I could read from it safely. But now it could throw an error in my runtime.

**Reply by tianzhich** [reply]
I'm using 6.12.1.

**Comment by bluebill1049** [comment]
can you share a codesandbox with your problem?

**Reply by tianzhich** [reply]
I'm sorry Bill there is no proper repro for this. I just wondering the `DeepPartial<TFieldValues>` below

```tsx
export declare type UseFormMethods<TFieldValues extends FieldValues = FieldValues> = {
  reset: (values?: UnpackNestedValue<DeepPartial<TFieldValues>>, omitResetState?: OmitResetState) => void;
}
```

Why do we need to partialize the fields? It makes a required field become not required. So I might forget to assign value to it when using `reset`. The typescript should give me some errors.

**Reply by bluebill1049** [reply]
what if some of the fields that users want to keep it the same?

**Reply by tianzhich** [reply]
what do you mean "keep it the same"? The same with what they defined? If I defined a required filed it should always be required. This what i expected.
